include/phasar/Utils/BitVectorSet.h

```diff
@@ -256,7 +269,21 @@ template <typename T> class BitVectorSet {
   [[nodiscard]] size_t size() const noexcept { return Bits.count(); }
 
   friend bool operator==(const BitVectorSet &Lhs, const BitVectorSet &Rhs) {
-    return Lhs.Bits == Rhs.Bits;
+    // Check, whether Lhs and Rhs actually have the same bits set and not
+    // whether their internal representation is exactly identitcal
```

Comment:
The problem is in the Mono analyses, where the `BitVectorSet` is used as container for dataflow facts. Especially, some of the flow functions remove (i.e. kill) facts from a `BitVectorSet` which may lead the last word in the set to become all-zero.
When comparing such a set with another that has the same bits set, but lacks the last (all-zero-) word, the original `operator==` would consider them different, although they contain exactly the same dataflow facts. This may lead to problems reaching the fixpoint in the solver.